src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/TelemetryReportingLSPRequestInvoker.cs

```diff
@@ -78,12 +80,16 @@ public TelemetryReportingLSPRequestInvoker(LSPRequestInvoker requestInvoker, ITe
         }
     }
 
-    private IDisposable? Track(string name, string method, string languageServerName)
+    internal IDisposable? Track(string name, string method, string languageServerName, Guid correlationId = default(Guid))
     {
+        if (correlationId == Guid.Empty)
+            return default;
+
         return _telemetryReporter.BeginBlock(name, Severity.Normal, ImmutableDictionary.CreateRange(new KeyValuePair<string, object?>[]
         {
             new("eventscope.method", method),
             new("eventscope.languageservername", languageServerName),
+            new("eventscope.correlationid", correlationId),
```

Comment:
Rather than duplicating these property names here, and in the diagnostics endpoint, it would be better to add a new method to `ITelemetryReport`, or perhaps an extension method on it, that does the job of this `Track` method. ie, It would be nicer if botht he diagnostics endpoint and the custom message target could just call `using var _ = _telemetryReporter.TrackLspRequest(name, method, servername, correlationId);` and not also need their own implementations of this method.